nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/cosmos/document/AzureCosmosSecureDBClientService.java

```diff
@@ -0,0 +1,158 @@
+)
+public class AzureCosmosSecureDBClientService
+        extends AbstractCosmosDBClientService
+        implements AzureCosmosDBConnectionService {
+
+    private String uriSecret;
+    private String accessKeySecret;
+    private String consistencyLevel;
+    private AzureKeyVaultConnectionService keyVaultClientService;
+
+    public static final PropertyDescriptor URI_SECRET = new PropertyDescriptor.Builder()
+            .name("azure-cosmos-db-uri-secret")
+            .displayName("Cosmos DB URI Secret Name")
+            .description("Cosmos DB URI Secret Name")
+            .required(true)
+            .addValidator(StandardValidators.URI_VALIDATOR)
+            .sensitive(true)
+            .build();
+
+    public static final PropertyDescriptor DB_ACCESS_KEY_SECRET = new PropertyDescriptor.Builder()
+            .name("azure-cosmos-db-key-secret")
+            .displayName("Cosmos DB Access Key Secret Name")
+            .description("Cosmos DB Access Key Secret Name")
+            .required(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .sensitive(true)
+            .build();
+
+    static final PropertyDescriptor KEYVAULT_CONNECTION_SERVICE = new PropertyDescriptor.Builder()
+            .name("azure-keyvault-connection-service")
+            .displayName("KeyVault Connection Service")
+            .description("If configured, the controller service used to obtain the secrets from KeyVault")
+            .required(true)
+            .identifiesControllerService(AzureKeyVaultConnectionService.class)
+            .build();
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        this.uriSecret = context.getProperty(URI_SECRET).getValue();
+        this.accessKeySecret = context.getProperty(DB_ACCESS_KEY_SECRET).getValue();
+        this.consistencyLevel = context.getProperty(
+                AzureCosmosDBUtils.CONSISTENCY).getValue();
+        this.keyVaultClientService = context.getProperty(
+                KEYVAULT_CONNECTION_SERVICE
+        ).asControllerService(AzureKeyVaultConnectionService.class);
+
+        createCosmosClient(
+                getURI(),
+                getAccessKey(),
+                getConsistencyLevel()
+        );
+    }
+
+    static List<PropertyDescriptor> descriptors = new ArrayList<>();
+
+    static {
+        descriptors.add(KEYVAULT_CONNECTION_SERVICE);
+        descriptors.add(URI_SECRET);
+        descriptors.add(DB_ACCESS_KEY_SECRET);
+        descriptors.add(AzureCosmosDBUtils.CONSISTENCY);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public String getURI() {
+        if (keyVaultClientService == null) {
+            throw new IllegalArgumentException(String.format(
+                    "Cannot get '%s'.", KEYVAULT_CONNECTION_SERVICE.getDisplayName()));
+        }
+        if (StringUtils.isBlank(uriSecret)) {
```

Comment:
Is it possible for `uriSecret` to be blank in light of the property validation and custom validate implementation?

Reply:
cleaned